app/server/lib/DocApi.ts

```diff
@@ -573,6 +573,19 @@ export class DocWorkerApi {
       })
     );
 
+    // Responds with an archive of all attachment contents, with suitable Content-Type and Content-Disposition.
+    this._app.get('/api/docs/:docId/attachments/download', canView, withDoc(async (activeDoc, req, res) => {
+      const archive = await activeDoc.getAttachmentsArchive(docSessionFromRequest(req));
+      res.status(200)
+        .type("application/zip")
+        // Construct a content-disposition header of the form 'attachment; filename="NAME"'
+        .set('Content-Disposition', contentDisposition(`${activeDoc.docName}.zip`, {type: 'attachment'}))
```

Comment:
Gave this a try. Everything worked well. The name was a bit jarring, compared to other downloads associated with a document. What do you think about a more human friendly name, that rhymes with what other `/download` methods do? https://github.com/gristlabs/grist-core/blob/14b91472aa3fa2f329aa1d02842ddbd36d93d785/app/server/lib/DocApi.ts#L1834-L1844

Something a bit weird seems to happen if I download with multiple distinct files that have the same name, but we've talked about changing the name structure so no need to poke at that...